### DISCUSSION TITLE
How to use GCP Datastream with Supabase?

### DISCUSSION THREAD

**Discussion by ydennisy** [supabase/supabase#9304] [Questions]
Google cloud have launched Datastream to push postgres data into bigquery: https://cloud.google.com/datastream/docs/configure-your-source-postgresql-database?hl=en_US#selfhostedpostgresql

This is very useful for merging transactional and analytics data.

Does anyone know if this is possible to use with supabase?

**Comment by ikhsan** [comment]
I recently migrated from Firebase to Supabase. Using Firebase, I was able to stream data to BigQuery using their [extensions](https://extensions.dev/extensions/firebase/firestore-bigquery-export). Now since I'm using Supabase, I don't know how to keep streaming my data to BigQuery.

I've followed their steps but I think some admin access wasn't allowed by default anymore. I've enquired Google support but no reply yet.

What I've done is follow Google's documentation on [self-managed postgres db](https://cloud.google.com/datastream/docs/configure-your-source-postgresql-database#selfhostedpostgresql), this particular line was failing

```sql
alter user <my_new_user> with replication;
# Failed to run sql query: must be superuser to alter replication roles or change replication attribute
```

Also, I'm unsure what's the default `wal_level` and either we can modify it or not.

What's confusing is that all validation passed

<img width="545" alt="image" src="https://user-images.githubusercontent.com/781254/213639807-fb2b196c-7092-44db-8c8e-92dadece8add.png">

but when creating the stream and start backfilling, it failed without much information

<img width="1166" alt="image" src="https://user-images.githubusercontent.com/781254/213639821-6cebe34c-6f41-4cee-9ed0-626c7562229c.png">

<img width="386" alt="image" src="https://user-images.githubusercontent.com/781254/213639836-57877c4d-3224-4107-999c-22720e91623f.png">


If anyone has successfully stream their supabase db, would love to understand this better :)
